File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> memo (nums.size(), 1);
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) memo[i] = max(memo[j] + 1, memo[i]);
            }
        }
        int res = 1;
        for (int i = 0; i < nums.size(); i++) {
            res = max(memo[i], res);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for num in nums:
            if num - 1 in s:
                continue
            i = 0
            while num + i in s:
                i+=1
                res = max(res, i)
        return res